### PR TITLE
Add buildargs

### PR DESCRIPTION
**Pull request by luiscoms**
- Add `$APP_TARGET` and `$MVN_ARGS` env var
- Fix tests for Mac when running `make test`

Thanks for @miklosbarabas for changes

### CODE REVIEW

**Comment by britter** [comment]
Wow, very nice! I'll need some time to test this change with my demo app. Thank you!

File: s2i/bin/run [hunk 1]
```diff
@@ -3,4 +3,6 @@
 set -e
 
 echo "---> Starting Spring Boot application"
-java -jar `find target -name *.jar`
+echo "--> # APP_TARGET = $APP_TARGET"
+echo "--> # JAVA_OPTS = $JAVA_OPTS"
+java $JAVA_OPTS -jar `find $APP_TARGET -name *.jar`
```
Comment (britter) [inline s2i/bin/run:10]: $APP_TARGET should be set to target by default if it is not set.

**Review by britter — CHANGES_REQUESTED** [review]

**Comment by luiscoms** [comment]
I just added target dir to default value

**Comment by britter** [comment]
@luiscoms when I run:

```
docker build -t codecentric/springboot-maven3-centos .
s2i build https://github.com/codecentric/springboot-sample-app codecentric/springboot-maven3-centos springboot-sample-app
```

I get:

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:46 min
[INFO] Finished at: 2016-09-20T06:58:00+00:00
[INFO] Final Memory: 31M/75M
[INFO] ------------------------------------------------------------------------
ERROR: chgrp: cannot access './.pki/nssdb': Permission denied
ERROR: chmod: cannot access './.pki/nssdb': Permission denied
ERROR: find: './.pki/nssdb': Permission denied

ERROR: An error occurred: non-zero (13) exit code from codecentric/springboot-maven3-centos
```

Does this work for you?

**Comment by luiscoms** [comment]
I Just run `make test`, that command build the image and run test-app

**Comment by luiscoms** [comment]
I Just run `make test`, and the tests are passing.
I ran your command and image built successfully

**Review by britter — APPROVED** [review]

**Comment by britter** [comment]
Thank you!

**Pull request merged**